server.go

```diff
@@ -545,15 +538,21 @@ func (srv *Server) serve(w *response, wg *sync.WaitGroup) {
 	wg.Done()
 }
 
-func (srv *Server) disposeBuffer(w *response) {
-	if w.udp != nil && cap(w.msg) == srv.UDPSize {
-		srv.udpPool.Put(w.msg[:srv.UDPSize])
+// Serve a new UDP request.
+func (srv *Server) serveUDPPacket(wg *sync.WaitGroup, m []byte, u *net.UDPConn, s *SessionUDP) {
+	w := &response{tsigSecret: srv.TsigSecret, udp: u, udpSession: s}
+	if srv.DecorateWriter != nil {
```

Comment:
this can be put in init as well?

Reply:
This is used to set a field on the `response` rather than the `Server` like the other ones did.

That being said, we could add a `srv.DecorateWriter = func(w Writer) Writer { return w }` and similar for `DecorateReader` in `Server.init`, though I'm not sure how clean that would end up being.

Reply:
ah, true, my mistake, I just saw `srv` and assumed.